### PR TITLE
Bug 1525444 - fix tests that poll for pulse messages

### PR DESCRIPTION
**Pull request by djmitche**
See bug for details..

Bugzilla Bug: [1525444](https://bugzilla.mozilla.org/show_bug.cgi?id=1525444)

### CODE REVIEW

**Comment by djmitche** [comment]
I'm working on those lats two checks.. and finding lots of small bugs to fix.

**Comment by djmitche** [comment]
Finally, all green!

**Comment by djmitche** [comment]
ping, anyone up for reviewing this? :)

It should fix the constant failing of the tc-queue tests in CI.

File: libraries/testing/README.md [hunk 1]
```diff
@@ -312,6 +312,39 @@ This function assumes the following config values:
...
+withPulse
+---------
+
+This function helps test applications that publish pulse messages.
+It is typically set up in `test/helper.js` like this:
+
+```js
+const {withPulse} = require('taskcluster-lib-testing');
+exports.withPulse = (mock, skipping) => {
+  withPulse({helper, skipping, namespace: 'taskcluster-someservice'});
```
Comment (imbstack) [inline libraries/testing/README.md]: Nice

File: libraries/testing/src/with-pulse.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    await helper.load('cfg');
+    client = new FakeClient(namespace);
+    helper.load.inject('pulseClient', client);
+
+    const matchingMessageExists = (exchange, check) =>
+      client.messages.some(message =>
+        (!exchange || message.exchange.endsWith(exchange)) &&
+        (!check || check(message)));
+
+    helper.assertPulseMessage = (exchange, check) => {
```
Comment (imbstack) [inline libraries/testing/src/with-pulse.js]: I notice that a lot of the checks we've replace used to have a `deepEqual`. I see that is not possible now with this more generic form but I wonder if we can have some way to at least log the payload of the message or something? Does not block landing this.

Comment (djmitche) [reply libraries/testing/src/with-pulse.js]: Yeah, that annoyed me at one point and I fell back to sticking in some `console.logs`.  `_.isEqual` works, but it doesn't give any diff when it fails.  I'm not sure how to fix that..

**Comment by djmitche** [comment]
rebased

Once that's green, I'll merge.

**Pull request merged**